### PR TITLE
created error message for failed projects api request with cypress test

### PR DESCRIPTION
**Pull request by destocot**
- created error message (with styles) for when the projects api request fails
- created cypress test to confirm error message upon failed api request
- created cypress test to confirm re-fetching of data upon clicking the "try again" button

### CODE REVIEW

**Review by mentor-tara — APPROVED** [review]
Go go go :smile:[](prolog-7)

File: features/projects/components/project-list/project-list.tsx [hunk 1]
```diff
@@ -3,15 +3,30 @@ import { useGetProjects } from "../../api/use-get-projects";
...
 
-  if (isLoading) {
+  if (isLoading || isRefetching) {
     return <div>Loading</div>;
   }
 
   if (isError) {
     console.error(error);
-    return <div>Error: {error.message}</div>;
+    return (
```
Comment (Jeloves) [inline features/projects/components/project-list/project-list.tsx:15]: This looks clean! My only suggestion is to move the code into a new "ErrorScreen" component that you could just import into "ProjectList" component.

File: features/projects/components/project-list/project-list.module.scss [hunk 2]
```diff
@@ -15,3 +17,39 @@
     grid-template-columns: repeat(auto-fit, 400px);
   }
 }
+
+.error {
+  background-color: color.$error-25;
+  padding: 16px;
```
Comment (Jeloves) [inline features/projects/components/project-list/project-list.module.scss:23]: Regarding padding, I think the way we're supposed to do it is by using values from @styles/space. The value for 16px would be "1rem".

Comment (Jeloves) [reply features/projects/components/project-list/project-list.module.scss:23]: Same thing for "gap" and "border-radius" (But to be honest with you, I'm still not totally sure which pixel values are supposed to be left as "pixels" vs using @styles/space).

File: cypress/e2e/project-list-error-message.cy.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      cy.wait(5000);
+    });
+
+    it("renders error message for failed projects request", () => {
+      cy.get("main")
+        .find("span")
+        .contains("There was a problem while loading the project data");
+    });
+
+    it("refetches data on click", () => {
```
Comment (Jeloves) [inline cypress/e2e/project-list-error-message.cy.ts:27]: These two it("") tests should be separate. The test from line 28 onward look good. But for it("refetches data on click"), what I did was basically copy the it("renders the projects") test that was originally written for us in project-list.cy

Comment (Jeloves) [reply cypress/e2e/project-list-error-message.cy.ts:27]: Sorry, I think I was wrong with the previous comment . From line 27 down, you test for the button click to refetch data, then you test to make sure that the refetching fails. I think that's good, but maybe you could include a test where the refetch is successful?

**Review by Jeloves — COMMENTED** [review]
Most of the code looks good! Here are my suggestions in one place:
- Create a new "Error" component, instead of writing HTML directly inside project-list
- Use styles/space.scss variables instead of pixels when it makes sense
- Include another cypress test where it tests for a successful refetch of data

* When I was leaving comments, I didn't realize I was supposed to highlight all the code I was referring to. So the code snippets below look weird. Sorry!

**Pull request merged**